openscap_report/report_generators/html_templates/js/oval_graph_generation_script.js

```diff
@@ -714,25 +739,60 @@ function generate_OVAL_error_message(test_info, div) {
     div.appendChild(BR.cloneNode());
 }
 
-function generate_referenced_endpoints(test_info, div) {
-    if (Object.keys(test_info.referenced_oval_endpoints).length > 0) {
-        const h1 = H1.cloneNode();
-        h1.textContent ='Referenced endpoints: ';
-        h1.className = "pf-c-title pf-m-lg";
-        div.appendChild(BR.cloneNode());
-        div.appendChild(h1);
-        for (const [id, endpoint] of Object.entries(test_info.referenced_oval_endpoints)) { // eslint-disable-line array-element-newline
+function add_header_referenced_endpoints_and_get_body(div) {
+    const container = DIV.cloneNode();
+    container.className = "pf-c-accordion pf-m-bordered";
+
+    const h3 = H3.cloneNode();
+    const button = BUTTON.cloneNode();
+    button.className = "pf-c-accordion__toggle pf-m-expanded";
+    button.setAttribute("type", "button");
+    button.setAttribute("aria-expanded","false");
+    button.addEventListener("click", show_OVAL_referenced_endpoints, false);
+    button.param_this = button;
+
+    const title = SPAN.cloneNode();
+    title.className = "pf-c-accordion__toggle-text";
+    title.textContent = "Referenced endpoints";
+
+    button.appendChild(title);
+    const icon = SPAN.cloneNode();
+    icon.className = "pf-c-accordion__toggle-icon";
+    const i = ICON.cloneNode();
+    i.className = "fas fa-angle-right";
+    i.setAttribute("aria-hidden", "true");
+    icon.appendChild(i);
+    button.appendChild(icon);
+
+    h3.appendChild(button);
+
+    const content = DIV.cloneNode();
+    content.className = "pf-c-accordion__expanded-content";
+    content.style.display = "none";
+    const body = DIV.cloneNode();
+    body.className = "pf-c-accordion__expanded-content-body";
+    content.appendChild(body);
+    button.param_content = content;
+
+    container.appendChild(h3);
+    container.appendChild(content);
+
+    div.appendChild(container);
+    return body;
+}
+
+function generate_referenced_endpoints(test_info, children_id, div) {
```

Comment:
Consider if the code complexity of this function can be reduced, for example by inverting the condition or extracting the inner for loop to a separate function.